File: algorithm/baekjoon/src/main/java/com/keunwon/algorithm/sorting/Problem1181.java
package com.keunwon.algorithm.sorting;

import java.io.*;
import java.util.*;

public class Problem1181 {

    public static void main(String[] args) throws IOException {
        final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine());
        Set<String> words = new HashSet<>();

        for (int i = 0; i < N; i++) {
            words.add(br.readLine());
        }

        StringBuilder sb = new StringBuilder();
        words.stream()
                .sorted((o1, o2) -> {
                    if (o1.length() == o2.length()) {
                        return o1.compareTo(o2);
                    }
                    return o1.length() - o2.length();
                })
                .forEach(s -> sb.append(s).append("\n"));

        bw.write(sb.toString());
        bw.flush();
    }
}

File: algorithm/kotlin-algorithm/src/main/kotlin/com/keunwon/algorithm/baekjoon/Problem1181.kt
package com.keunwon.algorithm.baekjoon

/**
 * Title: 단어 정렬
 * Level: 실버-5
 **/
class Problem1181 {
    fun solution(words: Array<String>): Array<String> {
        return words.sortedWith(compareBy({ it.length }, { it }))
            .distinct()
            .toTypedArray()
    }
}

fun main() {
    val n = readLine()!!.toInt()
    val words = Array(n) { readLine()!! }
    Problem1181().solution(words).forEach { println(it) }
}
